app/src/main/java/com/brainwallet/ui/BrainwalletActivity.kt

```diff
@@ -121,11 +164,47 @@ class BrainwalletActivity : BRActivity() {
                     }
 
                     is EventBus.Event.LegacyPasscodeVerified -> onPasscodeVerified(event.passcode)
+                    is EventBus.Event.LegacyUnLock -> onUnlock(event.passcode)
                 }
             }
             .launchIn(lifecycleScope)
     }
 
+
+    /**
+     * legacy logic, when the pin/passcode empty (not set)
+     * then should go to setpasscode
+     */
+    private fun onCheckPin() {
+        val pin = BRKeyStore.getPinCode(this)
+        if (pin.isEmpty()) {
```

Comment:
to force the users when already have 6 digit pin, we can add validation here `pin.length != 4`